src/frontend/screens/ProjectInviteModal/index.tsx

```diff
@@ -0,0 +1,273 @@
+import * as React from "react";
+import { Alert, StyleSheet, View } from "react-native";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+import { FormattedMessage, defineMessages, useIntl } from "react-intl";
+import { BottomSheetModal } from "@gorhom/bottom-sheet";
+
+import { MAPEO_BLUE, WHITE } from "../../lib/styles";
+import BottomSheet from "../../sharedComponents/BottomSheet";
+import Text from "../../sharedComponents/Text";
+import { DoneIcon } from "../../sharedComponents/icons";
+import Circle from "../../sharedComponents/icons/Circle";
+import Loading from "../../sharedComponents/Loading";
+import ModalActionButtons from "./ModalActionButtons";
+
+interface ProjectInviteDetails {
+  project: {
+    id: string;
+    name: string;
+  };
+  role: string;
+}
+
+type LoadingStatus = {
+  type: "loading";
+};
+
+type ErrorStatus = {
+  type: "error";
+  info: { error: Error };
+};
+
+type SuccessStatus = {
+  type: "success";
+  info: {
+    inviteDetails: ProjectInviteDetails;
+  };
+};
+
+const m = defineMessages({
+  title: {
+    id: "screens.ProjectInviteModal.title",
+    defaultMessage: "Join {projectName}",
+  },
+  inviteMessage: {
+    id: "screens.ProjectInviteModal.inviteMessage",
+    defaultMessage: "You've been invited to join\n {projectName} as a {role}",
+  },
+  declineInvite: {
+    id: "screens.ProjectInviteModal.declineInvite",
+    defaultMessage: "Decline Invite",
+  },
+  joinAndSync: {
+    id: "screens.ProjectInviteModal.joinAndSync",
+    defaultMessage: "Join And Sync",
+  },
+  close: {
+    id: "screens.ProjectInviteModal.close",
+    defaultMessage: "Close",
+  },
+  tryAgain: {
+    id: "screens.ProjectInviteModal.tryAgain",
+    defaultMessage: "Try Again",
+  },
+  noInviteFound: {
+    id: "screens.ProjectInviteModal.noInviteFound",
+    defaultMessage: "No Invite Found",
+  },
+});
+
+// Gregor TODO: Add method to API for decrypting invite
+const getProjectInviteDetails = async (
+  inviteKey: string
+): Promise<ProjectInviteDetails> => {
+  return new Promise((res, rej) => {
+    setTimeout(
+      () =>
+        Math.random() > 0.5
+          ? res({
+              project: {
+                name: `Project ${Math.floor(Math.random() * 1000)}`,
+                id: "123",
+              },
+              role: "Project Participant",
+            })
+          : rej(new Error("Could not get invite details")),
+      1000
+    );
+  });
+};
+
+const ProjectInviteModal: NavigationStackScreenComponent = ({ navigation }) => {
+  const { formatMessage: t } = useIntl();
+  const mountedRef = React.useRef(true);
+  const sheetRef = React.useRef<BottomSheetModal>(null);
+
+  const inviteKey: string | null = navigation.getParam("invite");
+
+  const [status, setStatus] = React.useState<
+    LoadingStatus | ErrorStatus | SuccessStatus
+  >(
+    inviteKey
+      ? { type: "loading" }
+      : { type: "error", info: { error: new Error(t(m.noInviteFound)) } }
```

Comment:
i don't think that's necessary in this case because this will only be run on initialization (i.e. even if `inviteKey` changes on rerenders). Providing the function option is more important if the value you're deriving from is 1) computationally expensive and 2) something that is only needed in the scope of the initialization (as opposed to the rest of the component)

in this case, getting `inviteKey` is probably not computationally expensive and it's also used elsewhere in the component, so it has to be computed at the component level and not just at the `useState` level